tools/build-toolchain.sh

```diff
@@ -1,34 +1,70 @@
+# N64 MIPS GCC toolchain build/install script (runs under UNIX systems).
 # (c) 2012-2021 Shaun Taylor and libDragon Contributors.
 # See the root folder for license information.
 
 
-
-# Before calling this script, make sure you have GMP, MPFR and TexInfo
-# packages installed in your system.  On a Debian-based system this is
-# achieved by typing the following commands:
+# Before calling this script, make sure you have all required
+# dependency packages installed in your system.  On a Debian-based systems
+# this is achieved by typing the following commands:
 #
-# sudo apt-get install libmpfr-dev
-# sudo apt-get install texinfo
-# sudo apt-get install libmpc-dev
+# sudo apt-get update && sudo apt-get upgrade
+# sudo apt-get install -yq wget bzip2 gcc g++ make file libmpfr-dev libmpc-dev zlib1g-dev texinfo git gcc-multilib
 
-# Exit on error
+# Exit script on error
 set -e
 
-# Set N64_INST before calling the script to change the default installation directory path
+# Ensure you set 'N64_INST' before calling the script to change the default installation directory path
+# by default it will presume 'usr/local'
 INSTALL_PATH="${N64_INST:-/usr/local}"
-# Set PATH for newlib to compile using GCC for MIPS N64 (pass 1)
-export PATH="$PATH:$INSTALL_PATH/bin"
+
+# Check for cross compile script flag
+if [ "$1" == "-xcw" ]; then # Windows cross compile flag is specified as a parameter.
+  # This (probably) requires the toolchain to have already built and installed on the native (linux) system first!
+  # This (may) also require the following (extra) package dependencies:
+  # sudo apt-get install -yq mingw-w64 libgmp-dev bison libz-mingw-w64-dev autoconf
+
+  echo "Cross compiling for different host"
+  # Use the current-directory/binaries for the install path, as this is not for linux!
+  rm -rf binaries && mkdir binaries # always ensure the folder is clean (if rebuilding)
+  THIS_SCRIPT_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+  FOREIGN_INSTALL_PATH="$THIS_SCRIPT_PATH/binaries"
+
+  # This will require the extra flags (under certain libs)
+  BUILD="--build=x86_64-linux-gnu"
+  HOST="--host=x86_64-w64-mingw32" 
+
+  # Dependency source libs (Versions)
+  # This will have to build Make and download FP libs
+  GMP_V=6.2.0
+  MPC_V=1.2.1
+  MPFR_V=4.1.0
+  MAKE_V=4.2.1
+
+else # We are compiling for the native system.
+  echo "building for native system"
+
+  # # Dependency source libs (Versions)
+  # BINUTILS_V=2.37
+  # GCC_V=11.2.0
+
+fi
+
+  # These "should" be the same as linux, but may be out of sync (as need to ensure working natively first).
+  # Binutils fails with 2.37 on canadian cross
+  BINUTILS_V=2.36.1 # so we are stuck with 2.36.1 for the moment
+  
+  # GCC 11.x fails on canadian cross
+  # see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100017
+  # see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80196
+  GCC_V=10.3.0 # so we are stuck with the 10.x branch for the moment.
```

Comment:
Can't you just download the patch found in Bugzilla and apply it using "patch" command?
https://gcc.gnu.org/bugzilla/attachment.cgi?id=51747

So that we build the same GCC for all targets

Reply:
I (personally) think that just makes things more complicated... What are the benefits of GCC 11.2 over 10.3 from a libdragon perspective... I like using the latest toolchain, but would rather wait for the next one than implement lots of workarounds...

Reply:
Consistency. I don't want to tell people that different toolchain on different OSs will generate different ROMs. So maybe I compile my game on Linux, then you compile the same game on Windows and yours doesn't work because of a GCC bug. Or doesn't compile because I'm using a command line option not available in your version. 

Every libdragon version should be compile with the single toolchain specified in the build script. If the build scripts creates different toolchains on different hosts, we can create problems. 

I'd rather make this script more complicated by applying a patch rather than pushing the problem of different toolchains to users.